### PR TITLE
chore: bump elastic-package to v0.32.1

### PR DESCRIPTION
**Pull request by mdelapenya**
- chore: bump elastic-package
- fix: use agent complete as fleet-server
- chore: bump go version to 1.17

## What does this PR do?
It bumps elastic-package to v0.32.1 (its latest version), and that forces us to bump the Go version to 1.17

On the same hand, because the changes in the elastic-package (see https://github.com/elastic/elastic-package/pull/637) are updating the base image for fleet-server, to use the elastic-agent complete flavour (including Synthetics and NodeJS runtime), we are using the very same image in the Docker Compose provisioner of this project.

## Why is it important?
Keep dependencies up-to-date

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## How to test this PR locally

Using docker provider:
```
TAGS="integrations" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false PROVIDER=docker make -C e2e/_suites/fleet functional-test
```

Using elastic-package provider:
```
TAGS="integrations" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false PROVIDER=elastic-package make -C e2e/_suites/fleet functional-test
```

## Related issues

- Relates https://github.com/elastic/beats/issues/29603
- Relates https://github.com/elastic/elastic-package/pull/637

## Follow-ups
We may need to enable dependabot for tracking changes in the elastic-package dependency :thinking:

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: .ci/ansible/playbook.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@
     - role: andrewrothstein.kind
     - role: gantsign.golang
       vars:
-        golang_version: 1.16.3
+        golang_version: 1.17
```
Comment (mdelapenya) [inline .ci/ansible/playbook.yml:38]: We need a way to pass `.go-version` file to here

**Review by adam-stokes — APPROVED** [review]

**Comment by mdelapenya** [comment]
@adam-stokes we are only using the elastic package provider in one test suite, right? Should we apply it everywhere?

**Comment by adam-stokes** [comment]
> @adam-stokes we are only using the elastic package provider in one test suite, right? Should we apply it everywhere?

I think that's a good idea

File: .ci/ansible/playbook.yml (hunk 1 above)
Comment (mdelapenya) [reply .ci/ansible/playbook.yml:38]: @adam-stokes what do you think about having a Make goal reading `.go-version` to install Go using Andrew Kroh's https://github.com/andrewkroh/gvm? Therefore, we won't duplicate Go versions in multiple places:

```shell
eval "$(gvm $(cat .go-version))"
```

**Pull request merged**